src/ApiException.php

```diff
@@ -138,9 +182,44 @@ public static function createFromRestApiResponse(
             $previous
         );
     }
+    
+    /**
+     * Checks if decoded metadata includes errorInfo message.
+     * @param array $decodedMetadata
+     * @return array $errorInfo {
+     *     @type boolean $containsErrorInfo
+     *     @type string $reason
+     *     @type string $domain
+     *     @type array $errorInfoMetadata
+     * }
+     */
+    public static function containsErrorInfo(array $decodedMetadata)
+    {
+        $errorInfo = [];
+        $errorInfo['containsErrorInfo'] = false;
+        if (empty($decodedMetadata)) {
+            return $errorInfo;
+        }
+        foreach ($decodedMetadata as $value) {
+            $isErrorInfoAny = array_key_exists('@type', $value) && ($value['@type'] === 'google.rpc.errorinfo-bin'
+                || $value['@type'] === 'type.googleapis.com/google.rpc.ErrorInfo');
+            $isErrorInfoArray = array_key_exists('reason', $value) && array_key_exists('domain', $value)
+                && array_key_exists('metadata', $value);
```

Comment:
Must an ErrorInfo have all fields populated? What if one field was omitted? I don't think metadata is required to be set, although I think domain and reason are.

Reply:
Confirmed with Summer that if ErrorInfo exists, all three fields should exist. But have updated function to just check for `reason` for simplicity.